### PR TITLE
first version

### PR DESCRIPTION
**Pull request by orthagonal**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+'use strict';
+const imagemin = require('imagemin');
+const imageminMozjpeg = require('imagemin-mozjpeg');
+const imageminPngquant = require('imagemin-pngquant');
+const Svgo = require('svgo');
+const fs = require('fs');
+const os = require('os');
+
+module.exports.compress = (file, level, callback) => {
```
Comment (jgallen23) [inline index.js]: Let's just export a single method that takes in the file or stream and some options and it figures out how to process automatically.  We could add support other file types too

File: index.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    }],
+  }, (err, results) => {
+    if (err) {
+      return callback(err);
+    }
+    callback(null, results.crop);
+  });
+};
+
+module.exports.compress = compress;
```
Comment (jgallen23) [inline index.js]: why do we need these broken out?

File: index.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  const pathToOutput = pathToFile;
+  gm(pathToFile)
+  .gravity(gravity)
+  .crop(dims[0], dims[1], origin[0], origin[1])
+  .write(pathToOutput, (err) => {
+    callback(err, pathToOutput);
+  });
+};
+
+module.exports.process = (imageFilePath, options, callback) => {
```
Comment (jgallen23) [inline index.js]: do we need a separate function?  couldn't it just be `require('optimiz')` and then `optimize(file, options)`

File: index.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+const imagemin = require('imagemin');
+const imageminMozjpeg = require('imagemin-mozjpeg');
+const imageminPngquant = require('imagemin-pngquant');
+const Svgo = require('svgo');
+const fs = require('fs');
+const os = require('os');
+const async = require('async');
+const path = require('path');
+
+const compress = (file, level, callback) => {
```
Comment (jgallen23) [inline index.js]: thoughts on moving each of the smaller functions into a lib directory?

Comment (orthagonal) [inline index.js]: If we're planning on expanding the things that optimiz can do, each operation should be in its own file.

File: index.js (hunk 2 above)
Comment (orthagonal) [inline index.js]: Thought it wouldn't hurt if people wanted to use them individually.  But you can get the same effect by just passing different options to 'process' anyway, so it's not really needed.

**Pull request merged**